src/tracer/ideal_age_example.F90

```diff
@@ -340,27 +374,122 @@ subroutine ideal_age_tracer_column_physics(h_old, h_new, ea, eb, fluxes, dt, G,
   year = US%s_to_T*time_type_to_real(CS%Time) * Isecs_per_year
 
   do m=1,CS%ntr
-    if (CS%sfc_growth_rate(m) == 0.0) then
-      sfc_val = CS%young_val(m)
+
+    if (CS%growth_rate(m) == 0.0) then
+      young_val = CS%young_val(m)
     else
-      sfc_val = CS%young_val(m) * &
-          exp((year-CS%tracer_start_year(m)) * CS%sfc_growth_rate(m))
+      young_val = CS%young_val(m) * &
+        exp((year-CS%tracer_start_year(m)) * CS%growth_rate(m))
     endif
-    do k=1,CS%nkml ; do j=js,je ; do i=is,ie
-      if (G%mask2dT(i,j) > 0.0) then
-        CS%tr(i,j,k,m) = sfc_val
-      else
-        CS%tr(i,j,k,m) = CS%land_val(m)
-      endif
-    enddo ; enddo ; enddo
-  enddo
-  do m=1,CS%ntr ; if (CS%tracer_ages(m) .and. &
-                      (year>=CS%tracer_start_year(m))) then
-!$OMP parallel do default(none) shared(is,ie,js,je,CS,nz,G,dt,Isecs_per_year,m)
-    do k=CS%nkml+1,nz ; do j=js,je ; do i=is,ie
-      CS%tr(i,j,k,m) = CS%tr(i,j,k,m) + G%mask2dT(i,j)*dt*Isecs_per_year
-    enddo ; enddo ; enddo
-  endif ; enddo
+
+    if (m == CS%BL_residence_num) then
+
+      if (CS%use_real_BL_depth) then
+        do j=js,je ; do i=is,ie
+          nk = floor(BL_layers(i,j))
+
+          do k=1,nk
+            if (G%mask2dT(i,j) > 0.0) then
+              CS%tr(i,j,k,m) = CS%tr(i,j,k,m) + G%mask2dT(i,j)*dt*Isecs_per_year
+            else
+              CS%tr(i,j,k,m) = CS%land_val(m)
+            endif
+          enddo
+
+          k = MIN(nk+1,nz)
+
+          if (G%mask2dT(i,j) > 0.0) then
+            layer_frac = BL_layers(i,j)-nk
```

Comment:
Why is layer_frac set at line 402, and then set again to the hard-coded value of 0.9 on line 403?  Is this a bug? Should one of these lines be commented out?

If both lines are intended to be there, please add a comment explaining this unusual construct.

Reply:
Thanks, @Hallberg-NOAA. We are waiting for a response from @sdbachman on this.